### DISCUSSION TITLE
Difference between predictor ran on local and the one ran on hugging face

### DISCUSSION THREAD

**Discussion by ep0p** [mindee/doctr#1629] [Q&A] [answered]
I have installed on local the tensorflow version of doctr. 
When running the ocr_predictor on local with these parameters ocr_predictor(det_arch='db_resnet50', reco_arch='crnn_vgg16_bn', pretrained=True) i can't seem to get all the text detected:
![Screenshot from 2024-06-04 11-50-15](https://github.com/mindee/doctr/assets/168719294/ae69e7ce-b261-40cd-8564-fce53e28acc0)

However, when i am running my document on the hugging face demo with the same models for det and rec, all the text gets detected:
![Screenshot from 2024-06-04 12-19-22](https://github.com/mindee/doctr/assets/168719294/cc13617c-64eb-4241-a3b8-3bfe88e4d8b6)

i really need to be able to detect all the text.

**Comment by felixdittrich92** [comment, marked as answer]
Hi @ep0p thanks for the report :)
Unfortunately the demo isn't up to date with the latest doctr version, but we have already a ticket to fix this

The huggingface demo uses pytorch as backend - could you give it a try with pytorch ?
CC @odulcy-mindee (#1090)

**Reply by ep0p** [reply]
You were right, pytorch version did it, thank you.

Should i understand that pytorch version is better at detecting and recognizing text or that is not at all the case?

**Reply by felixT2K** [reply]
With pytorch as backend it works better (all models) yes.
We try always to keep it comparable but sometimes the TF backend makes it hard for us 😅

**Reply by ep0p** [reply]
Alright, thanks again for your answer

**Comment by dlowe24** [comment]
Hi, I am trying the same thing on huggingface and Colab and it seems huggingface detect all text while Colab does not capture all text. 
I am using the same parameters and the latest doctr v0.9.0 with pytorch. See parameters below.

predictor = ocr_predictor(det_arch="db_resnet50",  # Text detection model
    reco_arch="crnn_vgg16_bn",assume_straight_pages=True, pretrained=True)

Is this still an issue ?

**Reply by felixdittrich92** [reply]
Hi @dlowe24 :wave:,

The HF demo is again out of date :sweat_smile: sry for this :pray: 
I opened now an automation to avoid this again https://github.com/mindee/doctr/pull/1742

**Reply by dlowe24** [reply]
Thanks. Could indicate which version huggingface was using ?

**Reply by felixT2K** [reply]
Should be 0.8.0 :)

**Reply by dlowe24** [reply]
thanks

**Reply by felixT2K** [reply]
Demo is now up to date with the main branch and will be updated automatically (also after each release) :)

**Comment by dlowe24** [comment]
Thanks. I can confirm the results are the same as in Colab. 
However, what ever model that was used in 0.8.0 seem to be a better model at recognizing all the text in the attached image.
![table_row_20](https://github.com/user-attachments/assets/4d0a7bb1-8018-4be3-8786-92bd967ad516)
